### PR TITLE
feat(connector): [Digital Virgo] template for integration

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

template code for digital virgo connector

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/common_enums/src/enums.rs [hunk 1]
```diff
@@ -209,6 +209,7 @@ pub enum RoutableConnectors {
     Cybersource,
     Datatrans,
     Deutschebank,
+    Digitalvirgo,
```
Comment (srujanchikke) [inline crates/common_enums/src/enums.rs:212]: comment this line

File: crates/api_models/src/enums.rs [hunk 2]
```diff
@@ -95,6 +95,7 @@ pub enum Connector {
     Cybersource,
     Datatrans,
     Deutschebank,
+    Digitalvirgo,
```
Comment (srujanchikke) [inline crates/api_models/src/enums.rs:98]: comment this line too

File: crates/api_models/src/enums.rs [hunk 3]
```diff
@@ -233,6 +234,7 @@ impl Connector {
             | Self::Coinbase
             | Self::Cryptopay
 			| Self::Deutschebank
+			| Self::Digitalvirgo
```
Comment (srujanchikke) [inline crates/api_models/src/enums.rs:237]: this too

File: crates/router/src/core/admin.rs [hunk 4]
```diff
@@ -1330,6 +1330,10 @@ impl<'a> ConnectorAuthTypeAndMetadataValidation<'a> {
                 deutschebank::transformers::DeutschebankAuthType::try_from(self.auth_type)?;
                 Ok(())
             }
+            api_enums::Connector::Digitalvirgo => {
+                digitalvirgo::transformers::DigitalvirgoAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
```
Comment (srujanchikke) [inline crates/router/src/core/admin.rs:1336]: this is not required for template code, can you comment this line ?

File: crates/router/src/types/transformers.rs [hunk 5]
```diff
@@ -262,6 +262,7 @@ impl ForeignTryFrom<api_enums::Connector> for common_enums::RoutableConnectors {
             api_enums::Connector::Cybersource => Self::Cybersource,
             api_enums::Connector::Datatrans => Self::Datatrans,
             api_enums::Connector::Deutschebank => Self::Deutschebank,
+            api_enums::Connector::Digitalvirgo => Self::Digitalvirgo,
```
Comment (srujanchikke) [inline crates/router/src/types/transformers.rs:265]: this too

**Review by srujanchikke — CHANGES_REQUESTED** [review]

**Review by srujanchikke — DISMISSED** [review]

File: crates/hyperswitch_connectors/src/connectors/digitalvirgo/test.rs [hunk 6]
```diff
@@ -0,0 +1,402 @@
+use masking::Secret;
+use router::{
+    types::{self, api, storage::enums,
+}};
+
+use crate::utils::{self, ConnectorActions};
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/connectors/digitalvirgo/test.rs:6]: Please rename this file to digitalvirgo.rs and move it to `crates/router/tests/connectors`

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by jarnura — APPROVED** [review]
On behalf of core approving the PR since there is no core changes.

**Pull request merged**